### PR TITLE
fix: update popover behavior for android

### PR DESCRIPTION
**Pull request by whitestranger7**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:
Fix for #1297

### CODE REVIEW

File: src/components/ui/popover/popover.component.tsx [hunk 1]
```diff
@@ -142,7 +148,16 @@ export class Popover extends React.Component<PopoverProps, State> {
...
+  public componentDidMount(): void {
+    if(Platform.OS === 'android') {
+      this.backHandler = BackHandler.addEventListener('hardwareBackPress', this.backAction);
+    }
+  }
+
   public componentWillUnmount(): void {
+    if(Platform.OS === 'android') {
+      this.backHandler.remove();
+    }
```
Comment (artyorsh) [inline src/components/ui/popover/popover.component.tsx:160]:
```suggestion
this.hardwareBackSubscription?.remove();
```

File: src/components/ui/popover/popover.component.tsx [hunk 2]
```diff
@@ -130,6 +131,11 @@ export class Popover extends React.Component<PopoverProps, State> {
     this.modalId = ModalService.hide(this.modalId);
   };
 
+  private backAction = (): boolean => {
```
Comment (artyorsh) [inline src/components/ui/popover/popover.component.tsx:134]:
```suggestion
  private onHardwareBackPress = (): boolean => {
```

File: src/components/ui/popover/popover.component.tsx [hunk 3]
```diff
@@ -99,6 +99,7 @@ export class Popover extends React.Component<PopoverProps, State> {
     forceMeasure: false,
   };
 
+  private backHandler: NativeEventSubscription;
```
Comment (artyorsh) [inline src/components/ui/popover/popover.component.tsx:102]:
```suggestion
  private hardwareBackSubscription: NativeEventSubscription;
```

**Review by artyorsh — CHANGES_REQUESTED** [review]
Let's use optional chaining to simplify code

**Comment by artyorsh** [comment]
Closes #1293

File: src/components/ui/popover/popover.component.tsx [hunk 4]
```diff
@@ -5,7 +5,7 @@
  */
 
 import React from 'react';
-import { StyleSheet } from 'react-native';
+import { StyleSheet, BackHandler, NativeEventSubscription, Platform } from 'react-native';
```
Comment (malashkevich) [inline src/components/ui/popover/popover.component.tsx:8]: Multiline

**Review by malashkevich — APPROVED** [review]

**Review by artyorsh — APPROVED** [review]

**Pull request merged**